proto/api.proto

```diff
@@ -67,6 +67,26 @@ enum ReadConsistencyType {
   INCONSISTENT = 2;
 }
 
+// A column key component. Multiple ColumnKey together are used to key into
+// a table index.
+message ColumnKey {
+  optional string column_name = 1;
+  optional bytes key = 2; 
+  // The end key is optional and not specified if only one column
+  // key is needed by the request. It must order strictly after key,
+  // and when specified, the request is for keys starting with key,
+  // up to but excluding end_key.
+  optional bytes end_key = 3; 
+}
+
+// A TableKey represents all the column key components used to key into
+// a table index.
+message TableKey {
+  optional string table_name = 1;
+  // The different key components from each column.
+  repeated ColumnKey columns = 2;
```

Comment:
I was imagining a `TableKey` would be composed of a table_name, an index name and the list of column values to populate the index key. Something like:

```
message TableKey {
  optional string table_name;
  optional string index_name;
  repeated bytes column_values;
}
```